### PR TITLE
Remove deprecated extends

### PR DESCRIPTION
**Pull request by ericfortmeyer**
See #152 and #147

### CODE REVIEW

File: src/js/index.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /* eslint-disable global-require */
 var preloadWebComponents = require('./preloadWebComponents'),
-    createChromecastButton = require('./components/ChromecastButton'),
-    createChromecastTech = require('./tech/ChromecastTech'),
+    { default: createChromecastButton } = require('./components/ChromecastButton'),
```
Comment (yokuze) [inline src/js/index.js:3]: Removing the mix of ESM and CommonJS will mean we won't have to do `{ default: createChromecastButton }` either.

**Review by yokuze — CHANGES_REQUESTED** [review]
@ericfortmeyer Thanks for the PR! Nice work. Just a few questions/suggestions for you before we can merge.

File: tests/ChromcastButton.test.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+const { default: createChromecastButton } = require('../src/js/components/ChromecastButton');
+
+class ButtonComponentStub {}
+
+describe('ChromecastButton', function() {
+   it('should not call videojs.extend', function() {
+      const videoJsSpy = {
+         extend: function() {
+            expect().fail('videojs.extends is deprecated');
```
[the remark this answers is not in the document]
Comment (ericfortmeyer) [reply tests/ChromcastButton.test.js]: Good catch. I fixed it.

File: package.json [hunk 3]
```diff
@@ -50,6 +50,7 @@
     "core-js": "3.11.0",
     "coveralls": "3.0.2",
     "eslint": "6.8.0",
+    "esm": "^3.2.25",
```
[the remark this answers is not in the document]
Comment (ericfortmeyer) [reply package.json]: I removed this package.  See comment below.

File: src/js/components/ChromecastButton.js [hunk 4]
```diff
@@ -1,183 +1,181 @@
...
-/**
-* The Video.js Button class is the base class for UI button components.
-*
-* @external Button
-* @see {@link http://docs.videojs.com/Button.html|Button}
-*/
-
-/** @lends ChromecastButton.prototype */
-ChromecastButton = {
+export default function createChromecastButton(videojs) {
```
[the remark this answers is not in the document]
Comment (ericfortmeyer) [reply src/js/components/ChromecastButton.js]: I reverted back to CommonJS.

File: src/js/index.js (hunk 1 above)
Comment (ericfortmeyer) [reply src/js/index.js]: See the comment above.

**Review by yokuze — APPROVED** [review]

**Pull request merged**